packages/block-library/src/categories/edit.js

```diff
@@ -106,14 +107,18 @@ class CategoriesEdit extends Component {
 	}
 
 	renderCategoryDropdown() {
-		const { showHierarchy } = this.props.attributes;
+		const { showHierarchy, instanceId, className } = this.props;
 		const parentId = showHierarchy ? 0 : null;
 		const categories = this.getCategories( parentId );
-
 		return (
-			<select className={ `${ this.props.className }__dropdown` }>
-				{ categories.map( ( category ) => this.renderCategoryDropdownItem( category, 0 ) ) }
-			</select>
+			<Fragment>
+				<label htmlFor={ `${ className }-${ instanceId }` } className="screen-reader-text">
+					{ __( 'Categories' ) }
+				</label>
+				<select id={ `${ className }-${ instanceId }` } className={ `${ className }__dropdown` }>
```

Comment:
I think `id` can be statically set in here to something like:
```js
const selectId = 'block-library-category';
...
<label htmlFor={ `${ selectId }-${ instanceId }` } className="screen-reader-text">
....
<select id={ `${ selectId }-${ instanceId }` } className={ `${ className }__dropdown` }>

Reply:
Thanks fora the feedback, I'll make this change and push a commit when I find some time later today.